ext/kv/sqlite.rs

```diff
@@ -212,30 +215,34 @@ impl<P: SqliteDbHandlerPermissions> DatabaseHandler for SqliteDbHandler<P> {
       }
     }
 
-    let conn = sqlite_retry_loop(|| {
+    let (conn, queue_waker_key) = sqlite_retry_loop(|| {
       let path = path.clone();
       let default_storage_dir = self.default_storage_dir.clone();
       async move {
         spawn_blocking(move || {
-          let conn = match (path.as_deref(), &default_storage_dir) {
-            (Some(":memory:"), _) | (None, None) => {
-              rusqlite::Connection::open_in_memory()?
-            }
-            (Some(path), _) => {
-              let flags =
-                OpenFlags::default().difference(OpenFlags::SQLITE_OPEN_URI);
-              rusqlite::Connection::open_with_flags(path, flags)?
-            }
-            (None, Some(path)) => {
-              std::fs::create_dir_all(path)?;
-              let path = path.join("kv.sqlite3");
-              rusqlite::Connection::open(path)?
-            }
-          };
+          let (conn, queue_waker_key) =
+            match (path.as_deref(), &default_storage_dir) {
+              (Some(":memory:"), _) | (None, None) => {
+                (rusqlite::Connection::open_in_memory()?, None)
+              }
+              (Some(path), _) => {
+                let flags =
+                  OpenFlags::default().difference(OpenFlags::SQLITE_OPEN_URI);
+                (
+                  rusqlite::Connection::open_with_flags(path, flags)?,
+                  Some(PathBuf::from(path)),
+                )
+              }
+              (None, Some(path)) => {
+                std::fs::create_dir_all(path)?;
+                let path = path.join("kv.sqlite3");
```

Comment:
The path needs to be resolved (relative to absolute, symlink resolution) before used as the key

Reply:
This path is already absolute. Why is the resolution needed?

Reply:
Sorry, this comment should be on the `(Some(path), _)` case - resolution is needed for that case because `path` is a user-provided string.

Reply:
Ah, thanks!